standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java

```diff
@@ -1283,6 +1307,23 @@ public Table getTable(String catName, String dbName, String tableName,
         rollbackTransaction();
       }
     }
+
+    // Get the statistics if requested and if the table can have actual data.
+    try {
+      if (tbl != null &&
+          !TableType.VIRTUAL_VIEW.toString().equals(tbl.getTableType()) &&
+          getColumnStats) {
+        tbl.setColStats(getTableColumnStatisticsInternal(catName, dbName, tableName,
+                 StatsSetupConst.getColumnStatsState(tbl.getParameters()), true, true));
+      }
+    }
+    catch (NoSuchObjectException nsoe) {
+      // By the time we fetch statistics we know that the table exists. So, we shouldn't get
```

Comment:
I am confused, there's already an object tbl which is not null, but the getStatistic() has failed, which can really never happen, so we will not throw that error any time. Ideally I would like ignore this exception completely. Is that possible? Returning a NULL object when we already have a non-NULL object in hand for a case which is supposed to happen never, is more confusing.

Reply:
It depends on timing. If the table is dropped before getting it for the first time, then we return null and it is not expected to return exception if table object is missing in first place. Also, it doesn't make sense to return the partially filled table object if we cannot get stats when caller expects it. So, I think, it is better to return null to adhere to current behaviour.

Reply:
I have made slight change in the way we add statistics to the table object. Instead of doing it in ObjectStore.java, I have done that in HiveMetaStore.java (taking inspiration from your idea that we should update statistics from HiveMetaStore.java instead of ObjectStore.java). That change addresses your comment and also reduced a lot of API changes to RawStore implementations. Further it allows cached statistics to be used in CachedStore when available and valid.

Please check the new implementation.